services/web/server/src/simcore_service_webserver/projects/_ports_api.py

```diff
@@ -141,18 +149,37 @@ class Config(PortLink.Config):
         allow_population_by_field_name = True
 
 
-def get_project_outputs(workbench: dict[NodeID, Node]) -> dict[NodeID, Any]:
-    """Returns values assigned to each output node"""
+class _OutputPortInfo(NamedTuple):
+    port_node_id: NodeID  # a probe node
+    task_node_id: NodeID  # a computational node
+    task_output_name: str  # an output of the computation node
```

Comment:
should this not be a IDStr? following your models in models_lib?

Reply:
and if you would use a dataclass, then you could add your comments as description fields.

Reply:
My goal here is to have a structure that is `tuple` but that is safe (e.g. that I do not mess up the order). I think NamedTuple are ideal for that. In this case, if I use `dataclass`, it would require me to use `as_tuple` to adapt to what I need. I do not think it is worth